### ISSUE TITLE
Huggingface Hub

### ISSUE THREAD

**Issue by AK391** [mindspore-ai/mindspore#178]
Hi, would you be interested in adding mindspore to Huggingface Hub ([Hugging Face Hub](https://huggingface.co/models))? The Hub offers free hosting, and it would make your work more accessible and visible to the rest of the ML community. I setup a example mindspore-ai organization where models, datasets, and spaces(web demos) can be added similar to the mindspore-ai org on github.

mindspore-ai: https://huggingface.co/mindspore-ai

Example from other organizations:
Keras: https://huggingface.co/keras-io
Microsoft: https://huggingface.co/microsoft
Facebook: https://huggingface.co/facebook

<img width="354" alt="Screen Shot 2022-02-11 at 10 24 34 AM" src="https://user-images.githubusercontent.com/81195143/153626675-8d3d44d5-84bd-448d-bcd5-52bd19407c95.png">

Here is how to integrate a library in the Hub: https://huggingface.co/docs/hub/adding-a-library

Integrating the Hub with your library provides many benefits, including:

- Free model hosting for you and your users.
- Built-in file versioning - even for huge files - made possible by [Git-LFS](https://git-lfs.github.com/).
- All public models are powered by the [Inference API](https://api-inference.huggingface.co/docs/python/html/index.html).
- In-browser widgets allow users to interact with your hosted models directly.

Please let us know if you would be interested and if you have any questions, we can also help with the technical implementation.

**Comment by huxiaoman7** [comment]
Hi, Thanks for your invitation, we would be very glad to add MindSpore to Huggingface Hub! I will read the doc you mentioned and try to integrate it in the Hub. Besides MindSpore, we also develop a high-level AI toolkit based on MindSpore, which we call it TinyMS: https://github.com/tinyms-ai. If you're interested in this project, we can also add it to Huggingface Hub.

**Comment by AK391** [comment]
@huxiaoman7 Great, can you please let me know your username of Huggingface so I can add you to the Mindspore-ai org or you can click the request to join this org button on the page https://huggingface.co/mindspore-ai. Then you can add models, dataset and spaces to the org in a similar way to github. Let me know if you have any questions

**Comment by huxiaoman7** [comment]
@AK391 Yes, Sure! My username is Charlotte77, and I also send the request, Thanks! :)

**Comment by AK391** [comment]
@huxiaoman7 thanks added, to start maybe we could add the quick start image classification example as a gradio space? similar to these keras examples https://huggingface.co/spaces/keras-io/vit-small-ds

**Comment by huxiaoman7** [comment]
@AK391 I've already create a pull request to add mindspore library to Huggingface Hub:https://github.com/huggingface/huggingface_hub/pull/684,  and waiting for approve. If it merge into main branch then you can add a quick start example.

**Comment by AK391** [comment]
@huxiaoman7 thank you, also adding guides here on adding spaces, models, and datasets to the mindspore-ai org on huggingface

How to add a Space: https://huggingface.co/blog/gradio-spaces
how to add models: https://huggingface.co/docs/hub/adding-a-model
uploading a dataset: https://huggingface.co/docs/datasets/upload_dataset.html

let me know if you have any questions

**Comment by AK391** [comment]
@huxiaoman7 great, I added the model for LeNet here https://huggingface.co/mindspore-ai/LeNet
for a good model card example see https://huggingface.co/keras-io/semantic-segmentation

regarding the PR, the only feedback currently is to add more models so users can click on the mindspore tag to link to the relevant models, thanks :)

**Comment by huxiaoman7** [comment]
@AK391 Thanks for adding models!! Today I'll add more models based on mindspore on huggingface:)

**Comment by osanseviero** [comment]
As I mentioned in the other PR, it would be great to have a couple more models before adding the official tag. Please also add a metadata tag by 
1. Creating a `README.md` file
2. Adding some content such as this

```
---
tags:
- mindspore
---

# Content
...
```

You can read more about model cards in https://huggingface.co/docs/hub/model-repos#what-are-model-cards-and-why-are-they-useful

cc @julien-c

**Comment by huxiaoman7** [comment]
@osansevieroThanks for kindly reminding me :), I've add a metadata tag as your instructions, you can see it in this model:https://huggingface.co/mindspore-ai/LeNet

**Comment by osanseviero** [comment]
This is great @AK391! :rocket:

**Comment by osanseviero** [comment]
@huxiaoman7, one question remaining is how to load these pretrained models. Do you have a code snippet showing how to load a model locally? I can then modify it to use a model from the Hub :smile:

**Comment by AK391** [comment]
@huxiaoman7 also note i get this error when using pip install mindspore==1.6.0

ERROR: Could not find a version that satisfies the requirement mindspore==1.6.0 (from versions: 1.2.0, 1.2.1, 1.3.0, 1.5.1)

https://pypi.org/project/mindspore/1.6.0/#history

**Comment by AK391** [comment]
@huxiaoman7 mindspore Lenet is trending on the hub 
<img width="353" alt="Screen Shot 2022-02-21 at 10 37 25 AM" src="https://user-images.githubusercontent.com/81195143/154986637-4596630e-518a-4f67-b6af-6ea206add42d.png">
🤗

**Comment by JIJIARONGjijiarong** [comment]
https://mindspore.cn/tutorials/en/r1.8/beginner/save_load.html   You can use load_checkpoint and load_param_into_net to load ckpt to your model.  @osanseviero

**Comment by JIJIARONGjijiarong** [comment]
pip install https://ms-release.obs.cn-north-4.myhuaweicloud.com/1.8.1/MindSpore/ascend/aarch64/mindspore_ascend-1.8.1-cp37-cp37m-linux_aarch64.whl --trusted-host ms-release.obs.cn-north-4.myhuaweicloud.com 
@AK391
